vcs/gitcmd/repo.go

```diff
@@ -277,10 +287,7 @@ func exitStatus(err error) int {
 	return 0
 }
 
-func (r *Repository) GetCommit(id vcs.CommitID) (*vcs.Commit, error) {
-	r.editLock.RLock()
-	defer r.editLock.RUnlock()
-
+func (r *Repository) getCommit(id vcs.CommitID) (*vcs.Commit, error) {
```

Comment:
Add a doc comment to `getCommit` saying that `r.editLock` MUST be held by the caller.